SQ-2825: Include new Samples MP into project.

shadeon opened the pull request:
This management pack includes fake "sample" management pack that can be imported to demonstrate what a variety of Enterprise applications could look like, and demonstrate SquaredUp EAM features.

All actual monitoring data is completely fake and is produced by a single cooked down PowerShell script that provides fixed health and random performance data we control. Availability monitoring looks real in the product but I've actually used the watcher override feature in the management pack to modify the web test scripts to produce canned data.

Note the huge number of singletons is to deal with a limitation within SquaredUp's EA loading code, that at this time only supports the loading of certain types of classes within discovery JSON map parsing routine.

No additional SCOM configuration is required to use this MP - upon import all sample EA instances will automatically begin discovery and start to populate with data.  All sample EAs are prefixed with "Sample - " in their display names (but are otherwise indistinguishable from standard EAs).

Note that the naming structure is a bit all over the place in the MP - this is because certain elements _have_ to follow the naming convention specified by SquaredUp's EA loading code. This is why you'll see GUIDs or some elements clearly using the wrong type prefix ("class" rather than "discovery") in their element Ids.

shadeon committed:
SQ-2825: Include new Samples MP into project.

This management pack includes fake "sample" management pack that can
be imported to demonstrate what a variety of Enterprise applications
could look like, and demonstrate SquaredUP EAM features.

WaynePlummer commented on ManagementPacks/SquaredUp.EAM.Samples/SampleEAs/AvailabilityOnly/Discovery.mpx:
```diff
@@ -0,0 +1,205 @@
+              </Setting>
+              <Setting>
+                <Name>$MPElement[Name='EAM!SquaredUp.EAM.Library.Class.AvailabilityMonitoring.Watcher']/ConfigurationJson$</Name>
+                <Value>{"url":"https://tailspintoys.com","match":"Tailspin","notmatch":"404"}</Value>
```
...so "https://tailspintoys.com" is one of MS's fictional outfits, right? It's interesting to see that there are now several real-looking toy resellers trading under that name!

WaynePlummer commented on ManagementPacks/SquaredUp.EAM.Samples/SampleEAs/AvailabilityWithDependencies/Discovery.mpx:
```diff
@@ -0,0 +1,231 @@
+          <DiscoveryClass TypeID="SquaredUp.EAM.Samples.Class.SampleCloudService" />
+        </DiscoveryTypes>
+        <DataSource ID="DS" TypeID="EAM!SquaredUp.EAM.Library.Datasource.DiscoveryProvider">
+          <DiscoveriesJson>
```
I *never* imagined anyone would ever end up writing one of these in a PR!!!!

WaynePlummer reviewed (approved):
What a lot of XML! I never imagined anyone would have to PR something with all this stuff in. Sorry there were so many naming conventions you were forced to adhere to (like, I said... I never imagined anyone except the EA authoring code would ever be doing this).

All looks entirely plausible... the proof of the pudding is if the correct stuff appears in the EA Designer and dashboards. Any chance of some screen-shots?

JRoseDev reviewed (approved):
Yep, that's definitely XML

shadeon committed:
SQ-2825: Added overrides for performance collection

Added overrides so availability monitoring performance collection
matches health state monitoring.

WaynePlummer commented on ManagementPacks/SquaredUp.EAM.Samples/Moduiles/FakeDataSource.mpx:
```diff
@@ -0,0 +1,70 @@
+
+        <DisplayString ElementID="SquaredUp.EAM.Samples.DataSource.FakeComputerData">
+          <Name>SquaredUp Sample EA fake computer data source</Name>
+          <Description>Produces fake data for health ahd performance collection</Description>
```
Typo

WaynePlummer commented on ManagementPacks/SquaredUp.EAM.Samples/SampleEAs/AvailabilityOnly/Discovery.mpx:
```diff
@@ -0,0 +1,205 @@
+      <DisplayStrings>
+
+        <DisplayString ElementID="SquaredUp.EAM.Samples.Discovery.AvailabilityOnly.AvailabilityMonitoringGroupContainsWatchers">
+          <Name>Discovers 'Sample - Tailspin Toys' Enterprise Application Availablity Monitoring Test instances</Name>
```
Typo

WaynePlummer commented on ManagementPacks/SquaredUp.EAM.Samples/SampleEAs/AvailabilityWithSimpleMap/Discovery.mpx:
```diff
@@ -0,0 +1,267 @@
+      <DisplayStrings>
+
+        <DisplayString ElementID="SquaredUp.EAM.Samples.Discovery.AvailabilityWithSimpleMap.AvailabilityMonitoringGroupContainsWatchers">
+          <Name>Discovers 'Sample - AdventureWorks OLTP' Enterprise Application Availablity Monitoring Test instances</Name>
```
Typo

WaynePlummer commented on ManagementPacks/SquaredUp.EAM.Samples/SampleEAs/CompleteEnterpriseApplication/Discovery.mpx:
```diff
@@ -0,0 +1,452 @@
+      <DisplayStrings>
+
+        <DisplayString ElementID="SquaredUp.EAM.Samples.Discovery.CompleteEnterpriseApplication.AvailabilityMonitoringGroupContainsWatchers">
+          <Name>Discovers 'Sample - Contoso Sales App' Enterprise Application Availablity Monitoring Test instances</Name>
```
Typo

WaynePlummer commented on ManagementPacks/SquaredUp.EAM.Samples/SampleEAs/AvailabilityWithDependencies/Discovery.mpx:
```diff
@@ -0,0 +1,231 @@
+      <DisplayStrings>
+
+        <DisplayString ElementID="SquaredUp.EAM.Samples.Discovery.AvailabilityWithDependencies.AvailabilityMonitoringGroupContainsWatchers">
+          <Name>Discovers 'Sample - SmartHotel 360' Enterprise Application Availablity Monitoring Test instances</Name>
```
Typo

WaynePlummer commented on ManagementPacks/SquaredUp.EAM.Samples/SampleEAs/AvailabilityOnly/Overrides.mpx:
```diff
@@ -0,0 +1,51 @@
+          <Name>Web Availability Test Override for 'Sample - Tailspin Toys'</Name>
+        </DisplayString>
+        <DisplayString ElementID="SquaredUp.EAM.Samples.Override.AvailabilityOnly.Availability.PerfCollection">
+          <Name>Web Availability Test Performance collection Override for 'Sample - Tailspin Toys'</Name>
```
Title Case?

WaynePlummer commented on ManagementPacks/SquaredUp.EAM.Samples/SampleEAs/AvailabilityWithDependencies/Overrides.mpx:
```diff
@@ -0,0 +1,51 @@
+          <Name>Web Availability Test Override for 'Sample - SmartHotel 360'</Name>
+        </DisplayString>
+        <DisplayString ElementID="SquaredUp.EAM.Samples.Override.AvailabilityWithDependencies.Availability.PerfCollection">
+          <Name>Web Availability Test Performance collection Override for 'Sample - SmartHotel 360'</Name>
```
Title Case?

WaynePlummer commented on ManagementPacks/SquaredUp.EAM.Samples/SampleEAs/AvailabilityWithSimpleMap/Overrides.mpx:
```diff
@@ -0,0 +1,51 @@
+          <Name>Web Availability Test Override for 'Sample - AdventureWorks OLTP'</Name>
+        </DisplayString>
+        <DisplayString ElementID="SquaredUp.EAM.Samples.Override.AvailabilityWithSimpleMap.Availability.PerfCollection">
+          <Name>Web Availability Test Performance collection Override for 'Sample - AdventureWorks OLTP'</Name>
```
Title Case?

WaynePlummer commented on ManagementPacks/SquaredUp.EAM.Samples/SampleEAs/CompleteEnterpriseApplication/Overrides.mpx:
```diff
@@ -0,0 +1,51 @@
+          <Name>Web Availability Test Override for 'Sample - Contoso Sales App'</Name>
+        </DisplayString>
+        <DisplayString ElementID="SquaredUp.EAM.Samples.Override.CompleteEnterpriseApplication.Availability.PerfCollection">
+          <Name>Web Availability Test Performance collection Override for 'Sample - Contoso Sales App'</Name>
```
Title Case?

shadeon committed:
SQ-2825: Fix typos in display strings
